tests/test_init.py

```diff
@@ -3,26 +3,196 @@
 
 """
 import pytest
+import datetime
 from bci_tester.data import INIT_CONTAINER
 from bci_tester.runtime_choice import DOCKER_SELECTED
 from pytest_container.runtime import LOCALHOST
 
+CONTAINER_IMAGES = [INIT_CONTAINER]
+
 
-@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
 @pytest.mark.skipif(
     DOCKER_SELECTED
     and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
     reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
 )
-def test_systemd_present(container):
-    """Check that :command:`systemctl` is in ``$PATH``, that :command:`systemctl
-    status` works and that :file:`/etc/machine-id` exists.
+class TestSystemd:
+    """
+    systemd test module for the bci-init container.
+    """
 
-    This test is currently broken due to `moby/moby#42275
-    <https://github.com/moby/moby/issues/42275>`_ with :command:`docker` and
-    :command:`systemd` >= 248.
+    def test_systemd_present(self, auto_container):
+        """Check that :command:`systemctl` is in ``$PATH``, that :command:`systemctl
+        status` works and that :file:`/etc/machine-id` exists.
 
-    """
-    assert container.connection.exists("systemctl")
-    assert container.connection.file("/etc/machine-id").exists
-    assert container.connection.run_expect([0], "systemctl status")
+        This test is currently broken due to `moby/moby#42275
+        <https://github.com/moby/moby/issues/42275>`_ with :command:`docker` and
+        :command:`systemd` >= 248.
+
+        """
+        assert auto_container.connection.exists("systemctl")
+        assert auto_container.connection.file("/etc/machine-id").exists
+        assert auto_container.connection.run_expect([0], "systemctl status")
+
+    def test_systemd_boottime(self, auto_container):
+        """Ensure the container startup time is reasonable"""
+
+        STARTUP_LIMIT = 5  # Container startup time limit in seconds
+        TARGET_LIMIT = (
+            5  # Limit for the systemd target to be reached in seconds
+        )
+
+        def extract_time(stdout, prefix):
+            """internal helper function to extract the time from systemd-analyze time
```

Comment:
I guess not but since it's an internal function also I'm wondering if that would be even desired.

The reason I left this within the `test_systemd_boottime` function is that this is a helper function that is useful exclusively in the scope of this function.